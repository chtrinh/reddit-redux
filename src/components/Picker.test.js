import React from 'react';
import Picker from './Picker';

it('renders without crashing', () => {
  const selectedSubreddit = 'frontend';
  const handleChange = jest.fn();

  const wrapper = shallow(
    <Picker
      value={selectedSubreddit}
      onChange={handleChange}
      options={['reactjs', 'frontend']}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it('should handle select change event', () => {
  const selectedSubreddit = 'frontend';
  const mockedEvent = { target: {} };
  const handleChange = jest.fn();

  const wrapper = shallow(
    <Picker
      value={selectedSubreddit}
      onChange={handleChange}
      options={['reactjs', 'frontend']}
    />
  );

  wrapper.find('select').simulate('change', mockedEvent);

  expect(handleChange).toHaveBeenCalled();
});
