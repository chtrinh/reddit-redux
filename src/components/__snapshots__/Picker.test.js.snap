// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<span>
  <h1>
    frontend
  </h1>
  <select
    onChange={[Function]}
    value="frontend"
  >
    <option
      value="reactjs"
    >
      reactjs
    </option>
    <option
      value="frontend"
    >
      frontend
    </option>
  </select>
</span>
`;
